src/components/navigation-header/index.tsx

```diff
@@ -1,158 +1,223 @@
+    icon: slug,
+    label: productSlugsToNames[slug],
+    path: `/${slug}`,
+  }))
+  const allMainMenuItems = [[homeMenuItem], betaProductMenuItems]
+  const productLogo = PRODUCT_SLUGS_TO_LOGOS[currentProduct.slug]
+  const isProductHomePage = currentPath === `/${currentProduct.slug}`
+
+  return (
+    <div className={s.leftSide}>
+      <div className={s.contentBeforeNav}>
+        <NavigationHeaderDropdownMenu
+          ariaLabel="Main menu"
+          buttonClassName={s.companyLogoMenuButton}
+          itemGroups={allMainMenuItems}
+          leadingIcon={companyLogo}
+        />
+        <Link href={`/${currentProduct.slug}`}>
+          <a
+            aria-current={isProductHomePage ? 'page' : undefined}
+            aria-label={`${currentProduct.name} home`}
+            className={s.productLogoLink}
+          >
+            <InlineSvg className={s.productLogo} src={productLogo} />
+          </a>
+        </Link>
+      </div>
+      {isBetaProduct && (
+        <nav className={s.nav}>
+          <ul className={s.navList}>
+            {PRODUCT_PAGE_NAV_ITEMS.map(
+              (navItem: {
+                id?: string
+                isSubmenu?: boolean
+                label: string
+                pathSuffix?: string
+              }) => {
+                const { id, isSubmenu, label, pathSuffix } = navItem
+                const ariaLabel = `${currentProduct.name} ${label}`
 
-  if (currentPath === '/') {
-    return (
-      <Header>
-        <div className={s.leftSide}>
-          {/* eslint-disable-next-line @next/next/no-img-element */}
-          <img
-            alt=""
-            className={s.logo}
-            src="https://via.placeholder.com/232x32?text=SITE-LOGO"
-          />
-          <nav className={s.nav}>
-            <ul className={s.navList}>
-              {homePageNavItems.map(
-                (navItem: { label: string; pathSuffix: string }) => {
-                  const { label, pathSuffix } = navItem
-                  const menuItems = betaProductSlugs.map(
-                    (slug: ProductSlug) => ({
-                      icon: slug,
-                      label: productSlugsToNames[slug],
-                      path: `/${slug}/${pathSuffix}`,
-                    })
+                const Submenu = () => {
+                  return (
+                    <NavigationHeaderDropdownMenu
+                      ariaLabel={ariaLabel}
+                      itemGroups={[
+                        currentProduct.navigationHeaderItems[id].map(
+                          ({ icon, label, pathSuffix }) => ({
+                            icon,
+                            label,
+                            path: `/${currentProduct.slug}/${pathSuffix}`,
+                          })
+                        ),
+                      ]}
+                      label={label}
+                    />
                   )
+                }
+
+                const NavLink = () => {
```

Comment:
These component definitions should be pulled out of the component body. The issue with this is that each time React renders, these components get re-created and so React thinks they are new component types, so it triggers a full re-mount. This would wipe any internal state and could cause tricky issues down the road.

(ref: https://hashicorp.slack.com/archives/GATJMHBCJ/p1649371041041179)

Reply:
With it mind that I plan to pull some other code out into a sub `components` directory in a follow-up PR, would you be okay with me moving these out at the same time as those? I don't want to bloat the file diff count of this PR too much, but I can make my next PR be for pulling out separately defined components if that works!

Reply:
Sure, find by me! 👍

Reply:
Awesome, thanks! I'll make sure to have that PR up for review on Monday. 😊